submission_offsets.py

```diff
@@ -0,0 +1,68 @@
+from flask.cli import FlaskGroup
+import click
+from collections import defaultdict
+
+import db
+import webserver
+
+from sqlalchemy import text
+
+cli = FlaskGroup(add_default_commands=False, create_app=webserver.create_app_flaskgroup)
+
+@cli.command(name='add-offsets')
+@click.option("--limit", "-l", default=10000)
+def add_offsets(limit):
+	"""Update lowlevel submission offsets with a specified limit."""
+	incremental_add_offset(limit)
+
+def incremental_add_offset(limit):
+	with db.engine.connect() as connection:
+		# Find the next batch of items to update
+		batch_query = text("""
```

Comment:
It looks to me like this script will only do 10000 items and then quit, is that right?
Instead, it should repeatedly do 10000 items until there are no more remaining. In a loop, get x items, then for each of them count the submission offsets and add them in a transaction.
Then, loop around again and get another x items. You can check the number of rows returned by the batch_result query, and when it's 0, quit.

Reply:
That’s correct, currently it only does 10000 and then quits. I understand what you’re saying. I’ll add the loop so that it completes all the batches one after another.